### PR TITLE
feat(front): implement sign up functionality with validation

### PR DESCRIPTION
**Pull request by GuillaumeKESTEMAN**
**Title:** Add sign-up page with form validation and backend integration

**Type of Pull Request:**

- [ ] Bug fix
- [x] New feature
- [ ] Documentation
- [ ] Other (specify):

**Associated Issue:**
N/A

**Context:**
This PR implements a complete sign-up functionality for the SlotFinder application. Users can now create new accounts through a dedicated sign-up page that validates input and communicates with the backend API for account creation.

**Proposed Changes:**
- **New SignUp Component**: Created a complete sign-up page with username, email, and password fields using React Hook Form and Yup validation
- **Form Validation**: Implemented comprehensive client-side validation including email format, password strength (letters, numbers, symbols), and minimum length requirements
- **API Integration**: Added `signUpApi` function and `useSignUp` hook to handle backend communication for account creation
- **Error Handling**: Implemented `getFormattedError` utility to parse and display server errors gracefully
- **Routing**: Added sign-up route (`/sign-up`) and navigation link from home page
- **Testing**: Created comprehensive test suite covering form validation, user interactions, and API error scenarios using Vitest, React Testing Library, and MSW
- **MSW Configuration**: Set up Mock Service Worker for API testing with proper request/response handling
- **Accessibility**: Ensured proper ARIA attributes, form labels, and error message associations for screen readers
- **Type Safety**: Added TypeScript types for sign-up request/response data structures
- **Environment Configuration**: Added development environment variable for backend URL

**Checklist:**

- [x] I have verified that my changes work as expected
- [x] I have updated the documentation if necessary
- [x] I have thought to rebase my branch
- [x] I have applied the correct label according to the type of PR (bug/feature/documentation)

**Additional Information:**
The implementation follows accessibility best practices with proper form labels, ARIA attributes, and error handling. Password validation requires a minimum of 8 characters with letters, numbers, and symbols. The form uses React Hook Form for optimal performance and user experience. All functionality is covered by unit tests including edge cases and error scenarios.

### CODE REVIEW

File: front/.env.development [hunk 1]
```diff
@@ -0,0 +1 @@
+FRONT_BACKEND_URL=http://localhost:3001
```
[the remark this answers is not in the document]
Comment (GuillaumeKESTEMAN) [reply front/.env.development:1]: Tu proposes quoi comme autre nom pour définir la route de base du back ?
